packages/react-ui-components/src/components/SchemaForm/Widgets/SelectWidget/SelectWidget.tsx

```diff
@@ -1,47 +1,72 @@
+			return searchDropdownOptions(searchTerm, defaultDropdownOptions);
+		}
+
+		return defaultDropdownOptions;
+	}, [searchTerm, defaultDropdownOptions]);
+	const [stateValue, setStateValue] = useState(processValue(schema, value));
 	const emptyValue = useMemo(() => (multiple ? [] : undefined), [multiple]);
+
+	const onSearchChange = useCallback(({ detail: searchedLabel }: CustomEvent<string>) => setSearchTerm(searchedLabel), []);
+	const onChangeOptionSelected = useCallback(({ detail: selectedOption }: CustomEvent<string>) => {
+		onChangeValue(selectedOption);
+	}, []);
+	const onChangeOptionsSelected = useCallback(({ detail: selectedOptionsMap }: CustomEvent<{ [key: string]: boolean }>) => {
+		const selectedOptions = getSelectedOptions(selectedOptionsMap);
+		onChangeValue(selectedOptions);
+	}, []);
+	const onSelectionCleared = useCallback(() => {
+		onChangeValue([]);
+	}, [onChange]);
+	const onChangeValue = useCallback(newValue => {
+		const processedValue = processValue(schema, newValue);
+		onChange(processedValue);
+		setStateValue(processedValue);
+	}, []);
+
 	const args = {
 		id,
 		label: label || schema.title,
 		placeholder: placeholder ? placeholder : undefined,
 		required,
 		disabled: disabled || readonly,
 		errorState: rawErrors.length > 0 ? EValidationState.Invalid : EValidationState.Valid,
-		displayValue: typeof value === 'undefined' ? emptyValue : value,
-		options: Array.isArray(enumOptions)
-			? enumOptions.reduce((acc, { value, label }) => {
-					const disabled = Array.isArray(enumDisabled) && enumDisabled.indexOf(value) != -1;
-					acc[value] = { value, label, disabled };
-					return acc;
-			  }, {})
-			: []
-	};
-
-	const [stateValue, setStateValue] = useState(processValue(schema, value));
-	const onChangeOptionSelected = ({ detail: selectedOption }: CustomEvent<string>) => {
-		const newValue = processValue(schema, selectedOption);
-		onChange(newValue);
-		setStateValue(newValue);
+		displayValue: typeof value === 'undefined' ? emptyValue : displayValue?.(value, enumOptions),
+		options: filteredOptions,
+		onSearchChange,
+		searchable
 	};
 
-	const onChangeOptionsSelected = ({ detail: selectedOptionsMap }: CustomEvent<{ [key: string]: boolean }>) => {
-		const selectedOptions = getSelectedOptions(selectedOptionsMap);
-		const newValue = processValue(schema, selectedOptions);
-		onChange(newValue);
-		setStateValue(newValue);
-	};
+	useEffect(() => {
+		setStateValue(processValue(schema, value));
+	}, [value]);
 
 	return (
 		<>
 			{!multiple && <KvSingleSelectDropdown {...args} selectedOption={stateValue} onOptionSelected={onChangeOptionSelected} />}
-			{multiple && <KvMultiSelectDropdown {...args} selectedOptions={buildSelectedOptions(stateValue)} onOptionsSelected={onChangeOptionsSelected} />}
+			{multiple && (
+				<KvMultiSelectDropdown
+					selectionClearable={selectionClearable}
+					onSelectionCleared={onSelectionCleared}
+					{...args}
```

Comment:
can we spread the rest after the last prop that's defined inline?

Reply:
I don't see any harm in changing that right now. In the future when we add the capability of override default props, this will need to change, but for now, let's move it below.